src/index.js

```diff
@@ -0,0 +1,215 @@
+    this.$racingCountForm.addEventListener(
+      'submit',
+      this.handleRacingCountFormSubmitEvent.bind(this)
+    );
+    this.$restartBtn.addEventListener('click', this.handleRestartBtnClickEvent.bind(this));
+  }
+
+  handleCarNameFormSubmitEvent(e) {
+    e.preventDefault();
+    const carNames = this.$carNamesInput.value;
+
+    if (this.isNotValidCarNames(carNames)) {
+      return;
+    }
+
+    const carNamesArray = carNames.split(RULES.CAR_NAME_SEPERATOR).map((carName) => carName.trim());
+
+    this.racingCarList = carNamesArray.map((carName) => new Car(carName));
+    modifyElementDisplayStyle(this.$racingCountForm, 'block');
+  }
+
+  isNotValidCarNames(carNames) {
+    if (carNames === RULES.EMPTY_STRING) {
+      handleError(ERROR_MESSAGES.EMPTY_CAR_NAME, this.$carNamesInput);
+    }
+
+    const carNamesArray = carNames.split(RULES.CAR_NAME_SEPERATOR).map((carName) => carName.trim());
+
+    if (carNamesArray.some((carName) => carName.length > RULES.MAX_CAR_NAME_LENGTH)) {
+      handleError(ERROR_MESSAGES.EXCEED_CAR_NAME_LENGTH, this.$carNamesInput);
+      return true;
+    }
+
+    if (carNamesArray.some((carName) => carName.length === RULES.ZERO_CAR_NAME_LENGTH)) {
+      handleError(ERROR_MESSAGES.BLANK_CAR_NAME, this.$carNamesInput);
+      return true;
+    }
+
+    return false;
+  }
+
+  handleRacingCountFormSubmitEvent(e) {
+    e.preventDefault();
+    const racingCount = this.$racingCountInput.value;
+
+    if (this.isNotValidRacingCount(racingCount)) {
+      return;
+    }
+
+    const racingCountNumber = convertToNumber(racingCount);
+    this.racingCount = racingCountNumber;
+    this.play();
+  }
+
+  isNotValidRacingCount(racingCount) {
+    if (racingCount === RULES.EMPTY_STRING) {
+      handleError(ERROR_MESSAGES.BLANK_RACING_COUNT, this.$racingCountInput);
+      return true;
+    }
+
+    const racingCountNumber = convertToNumber(racingCount);
+
+    if (typeof racingCountNumber !== 'number') {
+      handleError(ERROR_MESSAGES.NOT_NUMBER_TYPE, this.$racingCountInput);
+      return true;
+    }
+
+    if (isNotNaturalNumber(racingCountNumber)) {
+      handleError(ERROR_MESSAGES.NOT_NATURAL_NUMBER, this.$racingCountInput);
+      return true;
+    }
+
+    return false;
+  }
+
+  handleRestartBtnClickEvent(e) {
+    this.resetGameStatus();
+    this.resetElement();
+    this.modifyStyle();
+  }
+
+  resetGameStatus() {
+    this.racingCarList = [];
+    this.racingCount = 0;
+  }
+
+  resetElement() {
+    resetInputElementValue(this.$carNamesInput);
+    resetInputElementValue(this.$racingCountInput);
+    resetElementInnerText(this.$racingCarList);
+    resetElementInnerText(this.$finalWinnerResult);
+  }
+
+  modifyStyle() {
+    modifyElementDisplayStyle(this.$racingCountForm, 'none');
+    modifyElementDisplayStyle(this.$finalWinner, 'none');
+    modifyElementDisplayStyle(this.$restartSection, 'none');
+  }
```

Comment:
```suggestion
resetElement() {
    this.$carNamesInput = '';
    this.$racingCountInput = '';
    this.$racingCarList.innerText = '';
    this.$finalWinnerResult.innerText = '';
  }

  modifyStyle() {
    this.$racingCountForm.style.display = '';
    this.$racingCountForm.style.display = 'none';
    this.$finalWinner.style.display = 'none';
    this.$restartSection.style.display = 'none';
  }
```
지나친 추상화라고 생각합니다. 단순히 해당 요소를 직접 접근해서 값을 바꿔주는 일인데 함수로 추상화할 필요가 있을까요? 
1. 처음보는 사람에게는 임의의 함수 이름이 아니라 DOM 인터페이스 조작이 더 약속된 사용법이겠죠
2. 파라미터로 node를 넘기게 되면 잘못된 파라미터를 넘기게 될 수 있는 상황도 있습니다.
3. 콜스택에 함수를 할당하고 해제를 하는 비용이 있는데 굳이 비용을 더 들이면서 얻는 이점이 있을까요?

Reply:
노스가 링크 달아준 포스팅과 유사한 상황이네요.
코드의 간결화와 추상화에 지나치게 집착을 한 것 같네요.

1, 2, 3번 모두 동의합니다. 
지나친 추상화가 불러 일으킬 수 있는 부작용이군요.

해당 부분 수정하도록 하겠습니다.